src/Doctrine/Odm/Extension/ParameterExtension.php

```diff
@@ -45,7 +55,8 @@ private function applyFilter(Builder $aggregationBuilder, ?string $resourceClass
                 continue;
             }
 
-            $filter = $this->filterLocator->has($filterId) ? $this->filterLocator->get($filterId) : null;
+            $filter = $this->getFilter($filterId, $values);
```

Comment:
Why? It extracts complexity

Reply:
because at some point we'll not use the filter locator anymore, and we'll have to rollback this function. I'd prefer to introduce a new code path  for this, as it'll simplify the maintenance later on. 
Adding a new function increases complexity imo as we need to read the function to understand what it does.

Reply:
Done